### PR TITLE
Save resume game

### PR DESCRIPTION
**Pull request by lpenzey**
Game now autosaves progress if user quits mid-game. Upon starting the program again it will check for existence of a saved game and give user the option to load the saved game or start anew.

### CODE REVIEW

**Comment by ssunday** [comment]
^Looks like a rebasing issue. Maybe you branched off of master before the other PR was merge? `git rebase master` after running `git pull origin master` on master should fix it if there aren't bigger issues

**Comment by lpenzey** [comment]
@ssunday yeah that's what happened (rebase is done now)

File: lib/code.rb [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
-  
-  CodeLength = 1..4
-  ColorBank = 6
+	CodeLength = 1..4
+ 	ColorBank = 6
   
-  def generate_code
-    @colors = colors = ["r", "g", "o", "y", "b", "p"]
-    @answer = (CodeLength).map { @colors[rand(ColorBank)] }
+	def generate_code
```
Comment (lpenzey) [inline lib/code.rb]: not sure why the indent shows up like this here, it looks right in my text editor

**Comment by ssunday** [comment]
@lpenzey That's actually a merge commit, not a rebase per-say. Merge commits are uglier in the git history and a hard pass for me. Rebasing it basically puts your commits on top of the other branch....which would then make it so it would not have another commit in it.

File: lib/code.rb [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-  ColorBank = 6
+
+CODE_LENGTH = 1..4
+COLOR_BANK = 6
+COLORS = ["r", "g", "o", "y", "b", "p"]
   
-  def generate_code
-    @colors = colors = ["r", "g", "o", "y", "b", "p"]
-    @answer = (CodeLength).map { @colors[rand(ColorBank)] }
+	def generate_code
```
[the remark this answers is not in the document]
Comment (lpenzey) [reply lib/code.rb]: Yeah pairing would be great! This has happened a few times before where the code looks fine in the editor but gets mangled when I push to github so I'll look into RuboCop and TravisCl.

**Comment by ssunday** [comment]
👍 On rubocop and travis 👏 👏

**Pull request merged**